src/components/QuakeByMagnitude/QuakeByMag.jsx

```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { FetchQuakeMag, setMinMagnitude, setMaxMagnitude } from '../../Redux/QuakeSlice';
+import MagnitudeMap from './MagnitudeMap';
+
+const QuakeByMag = () => {
+  const dispatch = useDispatch();
+  const { earthquakes, isLoading, error, minMagnitude, maxMagnitude } = useSelector((state) => state.earthquake);
+
+  useEffect(() => {
+    dispatch(FetchQuakeMag());
+  }, [dispatch, minMagnitude, maxMagnitude]);
+
+  const handleMinMagnitudeChange = (e) => {
+    dispatch(setMinMagnitude(parseFloat(e.target.value)));
+  };
+
+  const handleMaxMagnitudeChange = (e) => {
+    dispatch(setMaxMagnitude(parseFloat(e.target.value)));
+  };
+
+  return (
```

Comment:
* And also the expected interactions for the details page is missing, kindly work on them as they are[ required ](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=the%20details%20page.-,Details%20page,on%20the%20%22Back%22%20button%20(%3C)%2C%20the%20user%20navigates%20to%20the%20home%20page.,-Testing%20requirements)
```
Details page

In the details page, the web app retrieves data from the API to show detailed data in the category.
When the user clicks on the "Back" button (<), the user navigates to the home page.
```